plugins/module_utils/foreman_helper.py

```diff
@@ -285,14 +295,132 @@ def ensure_resource(self, resource, entity_dict, entity, state=None, name_map=No
+        elif state == 'present':
+            if current_entity is None:
+                changed, current_entity = self._create_entity(resource, desired_entity, params, entity_spec)
+            else:
+                changed, current_entity = self._update_entity(resource, desired_entity, current_entity, params, entity_spec)
+        elif state == 'absent':
+            if current_entity is not None:
+                changed, current_entity = self._delete_entity(resource, current_entity, params)
+        else:
+            self.fail_json(msg='Not a valid state: {}'.format(state))
+        return changed, current_entity
+
+    def _create_entity(self, resource, desired_entity, params, entity_spec):
+        """Create entity with given properties
+
+            Parameters:
+                resource (string): Plural name of the api resource to manipulate
+                desired_entity (dict): Desired properties of the entity
+                params (dict): Lookup parameters (i.e. parent_id for nested entities) (optional)
+                entity_spec (dict): Description of the entity structure
+            Return value:
+                Pair of boolean indicating whether something changed and the new current state if the entity
+        """
+        payload = _flatten_entity(desired_entity, entity_spec)
+        if not self.check_mode:
+            if params:
+                payload.update(params)
+            return self._resource_action(resource, 'create', payload)
+        else:
+            fake_entity = desired_entity.copy()
+            fake_entity['id'] = -1
+            return True, fake_entity
+
+    def _update_entity(self, resource, desired_entity, current_entity, params, entity_spec):
+        """Update a given entity with given properties if any diverge
+
+            Parameters:
+                resource (string): Plural name of the api resource to manipulate
+                desired_entity (dict): Desired properties of the entity
+                current_entity (dict): Current properties of the entity
+                params (dict): Lookup parameters (i.e. parent_id for nested entities) (optional)
+                entity_spec (dict): Description of the entity structure
+            Return value:
+                Pair of boolean indicating whether something changed and the new current state if the entity
+        """
+        payload = {}
+        desired_entity = _flatten_entity(desired_entity, entity_spec)
+        current_entity = _flatten_entity(current_entity, entity_spec)
+        for key, value in desired_entity.items():
+            # query entity_spec[key] for special treatment
+            spec = entity_spec.get(key)
+            if spec:
+                # properties with 'force_update' must always be updated
+                if spec.get('force_update', False):
+                    payload[key] = value
+                    continue
```

Comment:
This is an example for values, that need a special treatment. Though i do not know whether `force_update` is really needed, because if the key is in `desired_entity` and not in `current_entity` (i.e. not reported by the server), it will end up in payload by the logic further down. Also i do not understand the conceptional difference between `force_update` and `check_missing`. So i guess, we don't need either of them.

Reply:
`check_missing` was introduced for Content View Filters, depending on the type of CVF we get a different set of attributes returned.  

https://github.com/theforeman/foreman-ansible-modules/blob/c92cf72879f765e98a2059291c2043b36ae5218d/plugins/module_utils/foreman_helper.py#L243-L246

https://github.com/Katello/katello/blob/master/app/views/katello/api/v2/content_view_filter_rules/show.json.rabl


I dont know the history of `force_update`

Reply:
I think, by looping over the user provided properties, we should end up including everything, the user wanted to have. But i guess/hope, we will see, whether it works when we get to the tests there.